scripting/src/test/java/org/infinispan/scripting/ScriptingTest.java

```diff
@@ -1,51 +1,205 @@
 package org.infinispan.scripting;
 
-import static org.testng.AssertJUnit.assertEquals;
-import static org.testng.AssertJUnit.assertNotNull;
-import static org.testng.AssertJUnit.assertTrue;
-
-import java.io.IOException;
-import java.io.InputStream;
-
 import org.infinispan.commons.CacheException;
-import org.infinispan.manager.EmbeddedCacheManager;
 import org.infinispan.tasks.TaskContext;
-import org.infinispan.test.SingleCacheManagerTest;
 import org.infinispan.test.TestingUtil;
-import org.infinispan.test.fwk.TestCacheManagerFactory;
-import org.testng.annotations.AfterMethod;
 import org.testng.annotations.Test;
-import static org.infinispan.scripting.ScriptingTests.loadScript;
 
-@Test(groups = "functional", testName = "scripting.ScriptingTest")
-public class ScriptingTest extends SingleCacheManagerTest {
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.Assert.assertNull;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
 
-   protected ScriptingManager scriptingManager;
+@Test(groups = "functional", testName = "scripting.ScriptingTest")
+public class ScriptingTest extends AbstractScriptingTest {
 
-   @Override
-   protected EmbeddedCacheManager createCacheManager() throws Exception {
-      return TestCacheManagerFactory.createCacheManager();
+   protected String[] getScripts() {
+      return new String[] { "test.js", "testMissingMetaProps.js", "testExecWithoutProp.js", "testInnerScriptCall.js" };
    }
 
    @Override
    protected void setup() throws Exception {
       super.setup();
-      scriptingManager = cacheManager.getGlobalComponentRegistry().getComponent(ScriptingManager.class);
    }
 
-
+   @Override
+   protected void clearContent() {
+      cacheManager.getCache().clear();
+   }
 
    public void testSimpleScript() throws Exception {
-      loadScript(scriptingManager, "test.js");
       String result = (String) scriptingManager.runScript("test.js", new TaskContext().addParameter("a", "a")).get();
       assertEquals("a", result);
    }
 
-   @Test(expectedExceptions = CacheException.class, expectedExceptionsMessageRegExp = "ISPN026005:.*")
+   @Test(priority = 2, expectedExceptions = CacheException.class, expectedExceptionsMessageRegExp = "ISPN026005:.*")
```

Comment:
It seems this test is tightly coupled with `testScriptingCacheClear` (because of the priority). Can we get rid of this dependency somehow?